"Non-square pixels" error raised for square pixels

Hi,

First of all, thanks for sharing your nice code with the community.

I have run successfully this example https://nbviewer.org/github/keflavich/pvextractor/blob/master/examples/IRAS05358Slicing.ipynb .

However, when I try to run it with my data, I obtain the following error:

> AssertionError                            Traceback (most recent call last)
> ~/.local/lib/python3.8/site-packages/pvextractor/utils/wcs_utils.py in get_spatial_scale(wcs, assert_square)
     > 19             np.testing.assert_almost_equal(abs(pc[0,0]), abs(pc[1,1]))
> ---> 20             np.testing.assert_almost_equal(abs(scale[0,0]), abs(scale[0,1]))
     > 21         except AssertionError:
> 
> /usr/lib/python3/dist-packages/numpy/testing/_private/utils.py in assert_almost_equal(actual, desired, decimal, err_msg, verbose)
    > 599     if abs(desired - actual) >= 1.5 * 10.0**(-decimal):
> --> 600         raise AssertionError(_build_err_msg())
    > 601 
> 
> AssertionError: 
> Arrays are not almost equal to 7 decimals
 > ACTUAL: 8.245905457097879e-05
 > DESIRED: 8.419853447744303e-05
> 
> During handling of the above exception, another exception occurred:
> 
> ValueError                                Traceback (most recent call last)
> <ipython-input-9-c281f7965c86> in <module>
      > 1 endpoints = [(10,10),(30,30)]
      > 2 xy = Path(endpoints)
> ----> 3 pv = extract_pv_slice(hdu, xy)
> 
> ~/.local/lib/python3.8/site-packages/pvextractor/pvextractor.py in extract_pv_slice(cube, path, wcs, spacing, order, respect_nan)
     > 76 
     > 77     if not isinstance(cube, np.ndarray) or wcs is not None:
> ---> 78         scale = get_spatial_scale(wcs)
     > 79         if isinstance(spacing, u.Quantity):
     > 80             pixel_spacing = (spacing / scale).decompose()
> 
> ~/.local/lib/python3.8/site-packages/pvextractor/utils/wcs_utils.py in get_spatial_scale(wcs, assert_square)
     > 20             np.testing.assert_almost_equal(abs(scale[0,0]), abs(scale[0,1]))
     > 21         except AssertionError:
> ---> 22             raise ValueError("Non-square pixels.  Please resample data.")
     > 23 
     > 24     return abs(scale[0,0]) * u.Unit(wcs.wcs.cunit[0])
> 
> ValueError: Non-square pixels.  Please resample data.

The spatial/celestial WCS information of my data is the following one:

> WCS Keywords
> 
> Number of WCS axes: 2
> CTYPE : 'RA---TAN'  'DEC--TAN'  
> CRVAL : 88.04742083333333  -7.456211111111111  
> CRPIX : 28.15333333333333  21.5  
> PC1_1 PC1_2  : 0.999945534290533  -0.0104368794387836  
> PC2_1 PC2_2  : 0.0104368794387836  0.999945534290533  
> CDELT : -8.3333333333333e-05  8.33333333333333e-05  
> NAXIS : 43  40

That is, CDELT values have the same size (with different sign to account for reflection), and PC values correspond to a rotation matrix, so I understand that my pixels are squares.

Based on this, I understand that there is a bug in the function `get_spatial_scale` in the file `utils/wcs_utils.py`, not?

Thanks for your help shedding light on this issue!

Hi, I have a related issue.  I'm not sure if it will be fixed by the PR, but I think the PR hasn't been merged, right?

I have the following code:
```
slice = PathFromCenter(center=SkyCoord(ra=source['pos_x'], dec=source['pos_y'], unit='deg'),
                       length=opt_view, angle=source['kin_pa']*u.deg, width=1*u.arcsec)
mask = fits.open(filename)
mask_pv = extract_pv_slice(mask[0].data, slice, wcs=WCS(mask[0].header, fix=True, translate_units='shd'))
```
And I get the error:
```
AssertionError: 
Arrays are not almost equal to 7 decimals
 ACTUAL: 0.0016675802169471255
 DESIRED: 0.0016657532823994607
```
The error has me stumped.  I don't know what I'm supposed to reproject to fix this since the WCS and the data come from the same fits files and the actual/desired really are almost equal.  Thanks in advance.
Also `CDELT1` and `CDELT2` in the `mask[0].header` are identical and equal to `1.6666670000E-3`

--- a/pvextractor/pvextractor.py
+++ b/pvextractor/pvextractor.py
@@ -13,7 +13,7 @@ from .utils.wcs_slicing import slice_wcs
 
 
 def extract_pv_slice(cube, path, wcs=None, spacing=1.0, order=3,
-                     respect_nan=True):
+                     respect_nan=True, assert_square=True):
     """
     Given a position-position-velocity cube with dimensions (nv, ny, nx), and
     a path, extract a position-velocity slice.
@@ -49,6 +49,11 @@ def extract_pv_slice(cube, path, wcs=None, spacing=1.0, order=3,
         the slices. If set to `True`, in the case of line paths a second
         computation is performed to ignore the NaN value while interpolating,
         and set the output values of NaNs to NaN.
+    assert_square : bool
+        If True, the WCS-loader will check whether the pixels are square.
+        If the pixels are not square, the interpretation of the X-axis in
+        the extracted PV diagram is ambiguous.  In some cases, it may be
+        necessary to disable this check, though.
 
     Returns
     -------
@@ -75,7 +80,12 @@ def extract_pv_slice(cube, path, wcs=None, spacing=1.0, order=3,
         wcs = sanitize_wcs(wcs)
 
     if not isinstance(cube, np.ndarray) or wcs is not None:
-        scale = get_spatial_scale(wcs)
+        try:
+            scale = get_spatial_scale(wcs, assert_square=assert_square)
+        except AssertionError as ex:
+            print("Pixels are non-square.  See error below.  You may "
+                  "disable this check by setting assert_square=False.")
+            raise ex
         if isinstance(spacing, u.Quantity):
             pixel_spacing = (spacing / scale).decompose()
             world_spacing = spacing

